src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -73,77 +75,59 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
+        if (!username.includes(usernameSuffix)) {
+            username += usernameSuffix;
         }
 
-        if (!username || !password) {
-            if (warnBeforePrompting) {
-                await ext.ui.showWarningMessage(
-                    localize('mustEnterUsernameAndPassword', 'You must enter the username and password for server "{0}" to continue.', this.parent.label),
-                    { modal: true },
-                    { title: localize('continue', 'Continue') }
-                );
-            }
-
-            username = await ext.ui.showInputBox({
-                prompt: localize('enterUsername', 'Enter username for server "{0}"', this.parent.label),
-                validateInput: (value: string) => { return (value && value.length) ? undefined : localize('usernameCannotBeEmpty', 'Username cannot be empty.'); }
-            });
-
-            const usernameSuffix: string = `@${this.parent.server.name}`;
-            if (!username.includes(usernameSuffix)) {
-                username += usernameSuffix;
-            }
-
-            password = await ext.ui.showInputBox({
-                prompt: localize('enterPassword', 'Enter password for server "{0}"', this.parent.label),
-                password: true,
-                validateInput: (value: string) => { return (value && value.length) ? undefined : localize('passwordCannotBeEmpty', 'Password cannot be empty.'); }
-            });
-
-            await this.persistServer(username, password);
-        }
+        const password: string = await ext.ui.showInputBox({
+            prompt: localize('enterPassword', 'Enter password for server "{0}"', this.parent.label),
+            password: true,
+            validateInput: (value: string) => { return (value && value.length) ? undefined : localize('passwordCannotBeEmpty', 'Password cannot be empty.'); }
+        });
 
+        await this.persistServer(username, password);
         return { username, password };
     }
 
-    private async getCredentialsFromKeytar(): Promise<{ username: string | undefined, password: string | undefined }> {
-        let username: string | undefined;
-        let password: string | undefined;
-
+    private async getCredentialsFromKeytar(): Promise<{ username: string, password: string }> {
         const storedValue: string | undefined = ext.context.globalState.get(this._serviceName);
         if (storedValue && this._keytar) {
             const servers: IPersistedServer[] = JSON.parse(storedValue);
             for (const server of servers) {
                 if (server.id === this._serverId) {
-                    username = server.username;
-                    password = await this._keytar.getPassword(this._serviceName, this._serverId) || undefined;
-                    break;
+                    return {
+                        username: server.username,
+                        password: <string>await this._keytar.getPassword(this._serviceName, this._serverId)
```

Comment:
Related to the above comment and type safety, it's not a great idea to cast this to `string`. For example - maybe someone looked through their keychain and deleted all passwords they didn't recognize - now this will return undefined/null and I feel like that should be treated the same way as if they didn't enter their password